File: python/evals/autocontext_fixture/src/db/queries.py
"""find_user_by_email + update_user_session."""
from src.config.database import get_db_connection
from src.db.models import User


def find_user_by_email(email: str):
    _conn = get_db_connection()
    # Stub — real impl would SELECT * FROM users WHERE email = ?.
    if email == "test@example.com":
        return User(1, email, "bcrypt$stub")
    return None


def update_user_session(user_id: int, session_token: str):
    _conn = get_db_connection()
    return True

File: python/evals/autocontext_fixture_rust/src/db/queries.rs
use crate::config::database::get_db_connection;
use crate::db::models::User;

pub fn find_user_by_email(email: &str) -> Option<User> {
    let _conn = get_db_connection();
    if email == "test@example.com" {
        Some(User { id: 1, email: email.into(), password_hash: "bcrypt$stub".into() })
    } else {
        None
    }
}

pub fn update_user_session(user_id: u64, token: &str) -> bool {
    let _conn = get_db_connection();
    let _ = (user_id, token);
    true
}
